SlicerLite/SlicerLiteLib/SlicerLiteModuleWidget.py

```diff
@@ -0,0 +1,216 @@
+import qt, slicer
+
+from SlicerLiteLib import Delegates, DataLoader, EventFilters, UIUtils, Settings, SlicerUtils, Model
+
+
+
+
+class SlicerLiteModuleWidget(qt.QWidget):
+    def __init__(self, parent=None):
+        super(SlicerLiteModuleWidget, self).__init__(parent)
+        qt.QVBoxLayout(self)
+        self.settings = qt.QSettings()
+        self.lastSelectedRowIndex = -1
+
+        self.dataLoader = DataLoader()
+        self.itemTableModel = Model.VolumeItemModel()
+        self.itemTableView = qt.QTableView()
+        self.deleteButtonItemDelegate = Delegates.DeleteButtonItemDelegate()
+        self.dicomTagsButtonItemDelegate = Delegates.DicomMetadataButtonItemDelegate()
+        self.deleteButtonItemDelegate.modelDeletedSignal.connect(self.onDeleteVolumeItem)
+
+        self.buttonsDelegate = [self.deleteButtonItemDelegate, self.dicomTagsButtonItemDelegate]
+
+        # Setup event filter
+        self.filter = EventFilters.DragAndDropEventFilter(slicer.util.mainWindow(), self.loadDicomDirectory)
+        slicer.util.mainWindow().installEventFilter(self.filter)
+
+        self.setupUI()
+
+    def setupUI(self):
+        """
+        Create and initialize UI components
+        """
+        self.setupTableViewLayout()
+        self.setupRenderingLayout()
+        self.setupSegmentationLayout()
+        self.layout().addStretch()
+
+    def setupTableViewLayout(self):
+        """
+        Setup the table view behavior/display
+        """
+        self.itemTableView.setItemDelegateForColumn(1, self.dicomTagsButtonItemDelegate)
+        self.itemTableView.setItemDelegateForColumn(2, self.deleteButtonItemDelegate)
+        # Hide headers
+        self.itemTableView.horizontalHeader().hide()
+        self.itemTableView.verticalHeader().hide()
+        # Hide table borders
+        self.itemTableView.setFrameStyle(qt.QFrame.NoFrame)
+        self.itemTableView.horizontalHeader().setSectionResizeMode(qt.QHeaderView.Stretch)
+        self.itemTableView.resizeColumnsToContents()
+        # Make lines no editabled
+        self.itemTableView.setEditTriggers(qt.QAbstractItemView.NoEditTriggers)
+        # Allow only one selection
+        self.itemTableView.setSelectionMode(qt.QAbstractItemView.SingleSelection)
+
+        self.itemTableView.clicked.connect(self.onTableViewItemClicked)
+        self.itemTableView.setModel(self.itemTableModel)
+
+        self.layout().addWidget(UIUtils.createButton("Load", callback=self.onClickLoadDicomDirectory))
+        self.layout().addWidget(qt.QLabel("Loaded volumes:"))
+        self.layout().addWidget(self.itemTableView)
+
+    def setupRenderingLayout(self):
+        """
+        Get and place shift rendering sliders for volume rendering
+        """
+        self.renderingModule = slicer.util.getNewModuleGui(slicer.modules.volumerendering)
+        self.shiftSliderWidget = slicer.util.findChild(self.renderingModule, "PresetOffsetSlider")
```

Comment:
Wondering if it's the best way to initialize the volume rendering.

In other projects, we use the qMRMLVolumePropertyNodeWidget directly and have utils to call the volume rendering logic without reusing the module widget.